### ISSUE TITLE
support 'as' prefixes for libs

### ISSUE THREAD

**Issue by subzero911** [google/json_serializable.dart#1360]
I import a helper library with "as" prefix
<img width="541" alt="image" src="https://github.com/google/json_serializable.dart/assets/12999702/7e38cc0a-6352-4d5e-8aa4-e02c66b60bd1">
And use it in annotations:
<img width="378" alt="image" src="https://github.com/google/json_serializable.dart/assets/12999702/58e15cf3-35d9-4b95-96b2-9b6300b33e82">

Generated code contains an error: 

<img width="705" alt="image" src="https://github.com/google/json_serializable.dart/assets/12999702/6985c1eb-37e6-4be2-8952-430740859c23">

**Comment by kevmoo** [comment]
Yup! An old issue. It's gnarly because it touches a LOT of code!

**Comment by woprandi** [comment]
It seems to be a common issue for any code generator

**Comment by kevmoo** [comment]
Yep! Tracking prefixing is tricky!
